### ISSUE TITLE
Game pauses instantly when starting beatmap

### ISSUE THREAD

**Issue by Theighlin** [ppy/osu#12644]
**Describe the bug:** Sometimes when i start up the game and start playing some maps, it always pauses as soon as i enter it and i can't press esc. Also, my monitor run at 66hz and usually the update runs at 280hz and draw at 140hz. It's the third time it has happened to me but instead of restarting the game i just alt-tabbed out and back into the game and everything was working as intended, also the sound increases in volume so i think the window might just be badly focused.

**Screenshots or videos showing encountered issue:** 

**osu!lazer version:** 2021.424.0

https://user-images.githubusercontent.com/53081750/116781453-4af5ea00-aa83-11eb-9ff9-9198093ff8e7.mp4

**Logs:** The video and the logs come from different game runs but the effect is the same as in the video.
 
[performance-draw.log](https://github.com/ppy/osu/files/6409493/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/6409494/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/6409495/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6409496/updater.log)
[database.log](https://github.com/ppy/osu/files/6409497/database.log)
[network.log](https://github.com/ppy/osu/files/6409498/network.log)
[performance.log](https://github.com/ppy/osu/files/6409499/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/6409500/performance-audio.log)

**Comment by bdach** [comment]
Going off the pause menu showing up and the fps capped at 60, sounds like a wrong window focus state problem.

**Comment by peppy** [comment]
@Theighlin did this start recently? also, can you still make it happen on the latest release?

**Comment by Theighlin** [comment]
I think it started recently, i've been playing lazer for a year now and don't recall this ever happening. Can still reproduce on latest version. I've managed to do it a couple of times by clicking on the desktop (maybe somewhere else too?) after launching the game but before the window opening up. Then when i click during the intro most of the times it updates correctly and rarely it doesn't but still gets input and i can play the game.

https://user-images.githubusercontent.com/53081750/116863534-bcaa7100-ac06-11eb-8b40-8d0284efe584.mp4

These are the logs that changed.

[performance.log](https://github.com/ppy/osu/files/6414213/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6414214/runtime.log)

**Comment by frenzibyte** [comment]
Can you try older releases and check which one doesn't have this issue?

**Comment by Theighlin** [comment]
I started from 2021.331.0 and whenever i launched the game and clicked on the desktop before it showed up, it was already focused as seen in the image below

![image](https://user-images.githubusercontent.com/53081750/116894767-b8944880-ac32-11eb-9f95-3dec0c0d3928.png)

from 2021.410.0, doing the same thing as before the game this time is not focused
![Screenshot_18](https://user-images.githubusercontent.com/53081750/116894787-c0ec8380-ac32-11eb-8587-2dbb1bbbd76d.png)

Even though it's not different from the ones above, here's another video

https://user-images.githubusercontent.com/53081750/116895206-3c4e3500-ac33-11eb-96dd-8c4b3012a33a.mp4

I tried launching it a couple of times on 2021.407.1 but since it never launched as unfocused i didn't try there anymore.

**Comment by PreciselyPixels** [comment]
I have this problem too, and it even extends to randomly pausing **during** a beatmap.

**Comment by peppy** [comment]
This is going to be the result of https://github.com/ppy/osu-framework/pull/4353. We'll need to investigate this further.

**Comment by peppy** [comment]
@PreciselyPixels / @Theighlin are you both able to test whether you can reproduce this in windowed or borderless mode? Also do you have anything weird running on your system (custom window managers or tools or anything)?

**Comment by PreciselyPixels** [comment]
> 
> 
> @PreciselyPixels / @Theighlin are you both able to test whether you can reproduce this in windowed or borderless mode? Also do you have anything weird running on your system (custom window managers or tools or anything)?

I went ahead and loaded Lazer today, but the problem fixed itself? I toggled the screen settings as you suggested and nothing brought the problem back. Weird.

**Comment by Theighlin** [comment]
> @PreciselyPixels / @Theighlin are you both able to test whether you can reproduce this in windowed or borderless mode? Also do you have anything weird running on your system (custom window managers or tools or anything)?

I can reproduce both in windowed and borderless, I don't have custom window managers or tools on my system nor discord or other overlays/socials running in the background.

**Comment by Theighlin** [comment]
Found out now that this also happens 100% of the times when unfocusing the game on startup and then start using only the tablet (in-game drivers). When mouse clicks seems to recover.

**Comment by frenzibyte** [comment]
Doesn't surprise me, tablets with in-game drivers are 100% decoupled from mouse and will not change focus state, but I'm interested to hear how you got it to run in unfocused state on fullscreen? I can reproduce the window being unfocused while in windowed, but not in fullscreen.

**Comment by peppy** [comment]
This seems like a UX oversight. A resolution would either be to make a tablet tap regain focus, or to disable tablet taps when the window doesn't have focus. I'm not sure which of these makes more sense so would be interested in hearing what the expected behaviour is from tablet users (likely the first will work fine in the majority of situations?)

Should be applied as a framework fix.

**Comment by GDcheeriosYT** [comment]
this problem is also happening to me, but I noticed how when I tabbed back in the fps cap was gone no pause on start of beatmap and my keyboard inputs didn't do anything ingame like searching for beatmaps, opening options, etc. I tabbed back in a third time now everything seems to be working

**Comment by peppy** [comment]
@GDcheeriosYT to clarify, are you also using a tablet and potentially leaving the game unfocused while inputting using the tablet alone? can you reproduce the issue if instead you use your mouse to navigate/play the game?

**Comment by Flutterish** [comment]
This is also sometimes manifests in osu!xr. The game is rendered on a different device than the window, and as such it is possible to click out of the window or minimize it, unfocusing it, which causes a headache in vr. I would propose you also adding a way for a gamehost to manage the focus state independently of the window (unless its already possible, but I havent been able to find it)

**Comment by peppy** [comment]
That is an unsupported scenario. The game must be focused to play it. Please keep it focused.

**Comment by peppy** [comment]
@Theighlin to confirm, does this still happen to you? we've made a lot of changes surrounding window handling (and so has SDL) so there's a potential that it has already been resolved.

**Comment by Theighlin** [comment]
I just booted up the game some 50 times or something, making sure it displayed everything at 60hz by clicking on the windows taskbar in the meanwhile and clicking back in at different times during the disclaimer. It always gained focus (2021.1207.0).
But my testing method wasn't the most reliable one back then so i downloaded a random version from may and actually managed to get this first try.
So i think this has since been fixed
